config/config.go

```diff
@@ -382,6 +382,10 @@ type ScrapeConfig struct {
 	MetricsPath string `yaml:"metrics_path,omitempty"`
 	// The URL scheme with which to fetch metrics from targets.
 	Scheme string `yaml:"scheme,omitempty"`
+	// The uncompressed response body size limit for scraping targets of this config.
+	// 0 means no limit, target uncompressed response body size more than this
+	// limit will cause the scrapes to fail.
+	BodySizeLimit int64 `yaml:"body_size_limit,omitempty"`
```

Comment:
We should have the unit somethere.


body_size_limit_bytes ?
or is there a proper way to take a size in go, like `3Gb`?

Reply:
I think `body_size_limit_bytes` would be simpler and better.  string like `3Gb` is human-readable but parser more complex. the other config field in prometheus are int or string or duration. 

etcd and loki also use `bytes`. https://github.com/etcd-io/etcd/blob/main/etcd.conf.yml.sample#L23 

https://github.com/grafana/loki/blob/main/cmd/loki/loki-local-config.yaml#L20

Reply:
@roidelapluie  I have done the revert 'move io.LimitReader to if branch' and rename body_size_limit to body_size_limit_bytes.  could you please review this PR again?

Reply:
We are using https://pkg.go.dev/github.com/alecthomas/units in cmd/prometheus. Can we reuse that library here?

Reply:
We can reuse that library function`n, err :=units.ParseBase2Bytes` to implement yaml unmarshaler interface，let me submit a new commit.

Reply:
I have use `units.Base2Bytes` in latest commit and rebase main branch.  @roidelapluie could you please review this PR again? 
![image](https://user-images.githubusercontent.com/7621241/119925980-0ccbe780-bfa9-11eb-9007-ad02bf78fd19.png)